cli/mfg.c

```diff
@@ -786,6 +788,191 @@ static int security_config_set(int argc, char **argv)
 	return 0;
 }
 
+#if HAVE_LIBCRYPTO
+static int load_sig_from_file(FILE *sig_fimg, uint8_t *sig)
+{
+	ssize_t rlen;
+
+	rlen = fread(sig, 1, SWITCHTEC_SIG_LEN, sig_fimg);
+
+	if (rlen < SWITCHTEC_SIG_LEN)
+		return -1;
+
+	return 0;
+}
+
+static int kmsk_already_exist(uint8_t *kmsk,
+			      struct switchtec_security_cfg_stat *state)
+{
+	int i;
+	int key_idx;
+	int match;
+
+	for(key_idx = 0; key_idx < state->public_key_num; key_idx++) {
+		match = 1;
+		for(i = 0; i < SWITCHTEC_KMSK_LEN; i++) {
+			if (state->public_key[key_idx][i] != kmsk[i]) {
+				match = 0;
+				break;
+			}
+		}
+
+		if (match)
+			return 1;
+	}
+
+	return 0;
+}
+#endif 
+
+static int kmsk_add(int argc, char **argv)
+{
+#if HAVE_LIBCRYPTO
```

Comment:
Better to put the #if outside the function and replicate the prototype. We don't like pre-processor directives mixed in code.